files/lib/data/conversation/ConversationAction.class.php

```diff
@@ -216,6 +216,11 @@ public function markAsRead() {
 		if (empty($this->parameters['visitTime'])) {
 			$this->parameters['visitTime'] = TIME_NOW;
 		}
+
+		// in case this is a call via PHP and the userID parameter is missing, set it to the userID of the current user
+		if (!array_key_exists('userID', $this->parameters)) {
```

Comment:
`array_key_exists()` will return true if `$this->parameters['userID'] === null`, but there is no legitimate case where this method should be invoked for non-existing users. Please change it into `!isset($this->parameters['userID'])`.

Also please do not open a new PR, I'll do a squash merge anyway.